### PR TITLE
fix: cleanup logs

### PR DESCRIPTION
**Pull request by trevorhibblen**
avoid logging PII in production environment
reduce log "noise" in production environment
throw exceptions to avoid "silent errors"

@schwamman Once everything looks good, we can rebuild dist after merge.

@ymarsal-kpa let me know if these changes make sense to you. Or if you need help understanding how of use debuglog instead of console.log.

Example:
```bash
NODE_DEBUG=log* node index.js
```

### CODE REVIEW

File: spectrum-integration/function/job/spectrum-user-job.ts [hunk 1]
```diff
@@ -32,87 +33,81 @@ export class SpectrumUserJob implements IJob {
...
+                }
 
-                    //Add User To List
-                    kpaUsers.push(kpaUser);
-                    status.upsertRecord++;
+                //Create Users
+                kpaUser.employeeNumber = `${companyCode.trim()}-${user.employeeCode.trim()}`;
+                kpaUser.firstName = user.firstName
+                kpaUser.lastName = user.lastName;
+                kpaUser.username = kpaUser.employeeNumber;
```
Comment (schwamman) [inline spectrum-integration/function/job/spectrum-user-job.ts:85]: Re-adding this change from last week. It got removed when merging in Yogi's branch.

```suggestion
                kpaUser.username = `${companyCode.trim()}-${user.employeeCode.trim()}`;
```

**Review by schwamman — CHANGES_REQUESTED** [review]
Looks good, I was able to build and run it locally with no problems. 

Just need to make the small change to the spectrum username.

File: spectrum-integration/function/job/spectrum-user-job.ts (hunk 1 above)
Comment (schwamman) [reply spectrum-integration/function/job/spectrum-user-job.ts:85]: You were correct in pulling in the latest change. I had merged in Yogis branch earlier though but it didn't have the latest updates from `main` pulled in. So it reverted just that one change.

https://github.com/HazardScout/anchorock-kpa/commit/86761802c8b347f49c6efa4cb411c2375b7034bc

**Review by ymarsal-kpa — APPROVED** [review]

**Review by schwamman — APPROVED** [review]
✅

**Pull request merged**